spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkDistributionAndOrderingUtil.java

```diff
@@ -163,7 +163,7 @@ private static Distribution positionDeleteDistribution(DistributionMode distribu
         return Distributions.clustered(clustering);
 
       case RANGE:
-        SortOrder[] ordering = new SortOrder[]{SPEC_ID_ORDER, PARTITION_ORDER, FILE_PATH_ORDER};
+        SortOrder[] ordering = new SortOrder[]{SPEC_ID_ORDER, PARTITION_ORDER, FILE_PATH_ORDER, ROW_POSITION_ORDER};
```

Comment:
I think including the row position may allow us to avoid straggler tasks when a particular file has most of deletes.